src/Components/WebAssembly/WebAssembly.Authentication/src/Interop/AuthenticationService.ts

```diff
@@ -282,16 +342,30 @@ export class AuthenticationService {
         return AuthenticationService.instance.signIn(state);
     }
 
-    public static completeSignIn(url: string) {
-        return AuthenticationService.instance.completeSignIn(url);
+    public static async completeSignIn(url: string) {
+        let operation = this._pendingOperations[url];
+        if (!operation) {
+            operation = AuthenticationService.instance.completeSignIn(url);
+            await operation;
+            this._pendingOperations[url] = undefined;
```

Comment:
```suggestion
            delete this._pendingOperations[url];
```

Not certain if this is what you want, but it would be leaking memory otherwise. I know it's hard to imagine someone do this so many times that would cause a problem.